### PR TITLE
Delete outdated files

### PR DESCRIPTION
**Pull request by kasohrab**
## Description
Title. For fun and historical reasons, I'll leave comments guessing why we previously used these files.

## Associated / Resolved Issue
Resolves [ClickUp card](https://app.clickup.com/t/2jmad9y)

### Test Case 1
1. make clean && make perf
2. run sim
3. no regressions

## Key Files to Review
basically just check if we need anyone of them still (I doubt it) or if there's anything else I should delete.
## Review Checklist

- [x] **Docstrings**: All methods and classes should have the file appropriate docstrings which follow the guidelines in the ["Contributing" page](https://rj-rc-software.readthedocs.io/en/latest/contributing.html) of our docs.
- [x] **Remove extra print statements**: Any print statements used for debugging should be removed
- [x] **Tag reviewers**: Tag some people for review and ping them on Slack

**Expected result:** works

### CODE REVIEW

File: autoupdate-docs.sh [hunk 1]
```diff
@@ -1,15 +0,0 @@
...
-
-set -e
-
-# This script is run by the CI setup to autoupdate our documentation
-
-git config --global user.name "$GIT_USERNAME"
-git config --global user.email $GIT_EMAIL
-mkdir -p api_docs
-git clone -b gh-pages git://github.com/robojackets/robocup-software api_docs/html
-make apidocs
```
Comment (kasohrab) [inline autoupdate-docs.sh:11]: we used to host docs on the gh-pages branch

File: cmake/MbedTargets.cmake [hunk 2]
```diff
@@ -1,1304 +0,0 @@
-#
-# ***** DO NOT MANUALLY EDIT THIS FILE *****
```
Comment (kasohrab) [inline cmake/MbedTargets.cmake:2]: robot previously used mbed as MCU.

File: cmake/MbedUtil.cmake [hunk 3]
```diff
@@ -1,284 +0,0 @@
...
-    if(NOT ${test})
-        message(FATAL_ERROR "Assertion failed: ${comment}")
-    endif()
-endmacro()
-
-
-# Set the mbed platform we should compile for.
-# Usage:
-#   mbed_set_platform(<var-name-prefixt> TARGET <target-id>)
-function(MBED_SET_PLATFORM target_var_base)
```
Comment (kasohrab) [inline cmake/MbedUtil.cmake:21]: firmware and software used to be a monorepo

File: robocup.sublime-project [hunk 4]
```diff
@@ -1,63 +0,0 @@
-{
```
Comment (kasohrab) [inline robocup.sublime-project:1]: people liked sublime

File: runcomp.sh [hunk 5]
```diff
@@ -1,29 +0,0 @@
-#!/bin/bash
-
-if ! command -v git >/dev/null; then
```
Comment (kasohrab) [inline runcomp.sh:3]: from before ros2, custom launch script

File: util/arch-setup [hunk 6]
```diff
@@ -1,41 +0,0 @@
...
-# specified in the environment if available, otherwise fallback to a default. To
-# run arch-setup with a different aur helper, do:
-#   AUR_HELPER=myprogram ./util/arch-setup
-# note that this only works with 'pacman-like' aur helpers.  See
-# https://wiki.archlinux.org/index.php/AUR_helpers#Comparison_table for more info
-AUR_HELPER=${AUR_HELPER:-yaourt}
-
-BASE=$(readlink -f $(dirname $0)/..)
-
-echo "-- Installing grSim default configuration"
```
Comment (kasohrab) [inline util/arch-setup:13]: people liked arch

File: util/osx-setup [hunk 7]
```diff
@@ -1,52 +0,0 @@
-#!/bin/bash
-
-# abort script if something fails
-set -e
-
-BASE="$(dirname $0)/.."
-
-if [ ! $(which brew) ]; then
```
Comment (kasohrab) [inline util/osx-setup:8]: before ros2 we could support osx.

File: util/profile [hunk 8]
```diff
@@ -1,43 +0,0 @@
-#!/bin/bash -e
-
-BASE=$(basename $1)
-
-# Directory containing this script
-UTIL_DIR=$(readlink -f $(dirname $0))
-
-echo "Running $*"
-
-# We have this mess because there is no way to get the PID
```
Comment (kasohrab) [inline util/profile:10]: no idea lol

File: cmake/FindBullet.cmake [hunk 9]
```diff
@@ -1,77 +0,0 @@
-# - Try to find the Bullet physics engine
```
Comment (kasohrab) [inline cmake/FindBullet.cmake:1]: we used to have our own simulator, then we switched to grSim (now we use erforce's sim). Our 2012 and 2013 [tdps](https://wiki.robojackets.org/Team_Description_Papers) talk about it a good bit

File: soccer/CMakeLists.txt [hunk 10]
```diff
@@ -37,9 +37,6 @@ find_package(PythonLibs 3.2 REQUIRED)
 # Eigen - used for linear algebra
 find_package(Eigen3 REQUIRED)
 
-# libusb
-find_package(libusb-1.0 REQUIRED)
```
Comment (p-nayak11) [inline soccer/CMakeLists.txt:41]: I assume that we don't need to communicate with a USB anymore?

**Review by p-nayak11 — CHANGES_REQUESTED** [review]
All of the deletions make sense. I believe the rj_gameplay/tests folder is something that should be considered for scrapping as well since we don't use it at all (unless it has a purpose?).

**Comment by kasohrab** [comment]
> All of the deletions make sense. I believe the rj_gameplay/tests folder is something that should be considered for scrapping as well since we don't use it at all (unless it has a purpose?).

I deleted all the tests that are outdated, the test-rc.py is still useful. I'm not sure if our CI actually runs that test right now 😅, but still it's a good regression test to have. We can always design a good testing framework later, so I see no issue with deleting the outdated stuff right now.

File: soccer/CMakeLists.txt [hunk 11]
```diff
@@ -56,39 +53,6 @@ find_package(
         COMPONENTS system
         REQUIRED)
 
-# find boost python This package is named 'python3' on Arch, 'python-py34' on Ubuntu 14, and
-# 'python' on OS X
-set(FOUND_A_BOOST_PYTHON FALSE)
```
Comment (kasohrab) [inline soccer/CMakeLists.txt:61]: we use ros2 now, and no longer need boost python

**Review by p-nayak11 — APPROVED** [review]
LGTM! I think referring this PR in the docs log or somewhere else would be a good idea to keep track of how the codebase is changing while we are at GT.

**Review by kfu02 — APPROVED** [review]
can't believe we used to have firmware sounds

**Pull request merged**